modules/linear_boltzmann_solvers/discrete_ordinates_solver/sweep/communicators/aah_async_comm.cc

```diff
@@ -44,228 +44,201 @@ AAH_ASynchronousCommunicator::ClearLocalAndReceiveBuffers()
+      delayed_prelocI_message_blockpos_[prelocI].push_back(pre_block_pos);
+      delayed_prelocI_message_size_[prelocI].push_back(num_unknowns);
     }
 
-    delayed_prelocI_message_received.emplace_back(message_count, false);
+    delayed_prelocI_message_received_.emplace_back(message_count, false);
   }
 
   // Successor locations
   size_t num_successors = spds.GetLocationSuccessors().size();
 
-  deplocI_message_count.resize(num_successors, 0);
-  deplocI_message_size.resize(num_successors);
-  deplocI_message_blockpos.resize(num_successors);
+  deplocI_message_count_.resize(num_successors, 0);
+  deplocI_message_size_.resize(num_successors);
+  deplocI_message_blockpos_.resize(num_successors);
+  deplocI_message_request_.clear();
 
-  deplocI_message_request.clear();
-
-  for (int deplocI = 0; deplocI < num_successors; deplocI++)
+  for (int deplocI = 0; deplocI < num_successors; ++deplocI)
   {
-    u_ll_int num_unknowns = aah_fluds.GetDeplocIFaceDOFCount(deplocI) * num_groups_ * num_angles_;
-
-    u_ll_int message_size;
-    int message_count;
-    if ((num_unknowns * 8) <= EAGER_LIMIT)
+    size_t num_unknowns = fluds.GetDeplocIFaceDOFCount(deplocI) * num_groups_ * num_angles_;
+    size_t message_count = num_angles_;
+    if (num_unknowns * 8 > max_mpi_message_size_)
+      message_count = ((num_unknowns * 8) + (max_mpi_message_size_ - 1)) / max_mpi_message_size_;
+    size_t message_size = (num_unknowns + (message_count - 1)) / message_count;
+
+    deplocI_message_count_[deplocI] = message_count;
+    deplocI_message_size_[deplocI].reserve(message_count);
+    deplocI_message_blockpos_[deplocI].reserve(message_count);
+
+    size_t dep_block_pos = 0;
+    for (int m = 0; m < message_count - 1; ++m)
     {
-      message_count = static_cast<int>(num_angles_);
-      message_size = ceil((double)num_unknowns / (double)message_count);
-    }
-    else
-    {
-      message_count = ceil((double)num_unknowns * 8 / (double)(double)EAGER_LIMIT);
-      message_size = ceil((double)num_unknowns / (double)message_count);
-    }
-
-    deplocI_message_count[deplocI] = message_count;
-
-    u_ll_int dep_block_pos = 0;
-    for (int m = 0; m < (message_count - 1); m++)
-    {
-      deplocI_message_size[deplocI].push_back(message_size);
-      deplocI_message_blockpos[deplocI].push_back(dep_block_pos);
+      deplocI_message_size_[deplocI].push_back(message_size);
+      deplocI_message_blockpos_[deplocI].push_back(dep_block_pos);
       num_unknowns -= message_size;
       dep_block_pos += message_size;
     }
     if (num_unknowns > 0)
     {
-      deplocI_message_blockpos[deplocI].push_back(dep_block_pos);
-      deplocI_message_size[deplocI].push_back(num_unknowns);
+      deplocI_message_blockpos_[deplocI].push_back(dep_block_pos);
+      deplocI_message_size_[deplocI].push_back(num_unknowns);
     }
 
-    deplocI_message_request.emplace_back(message_count);
+    deplocI_message_request_.emplace_back(message_count, MPI_Request());
```

Comment:
Technically, `mpi::Request()` to match the underlying type.

Reply:
Fixed.